core/tx_pool.go

```diff
@@ -241,13 +245,20 @@ func (pool *TxPool) Pending() map[common.Address]types.Transactions {
 	pool.mu.Lock()
 	defer pool.mu.Unlock()
 
+	pending := make(map[common.Address]types.Transactions)
+
+	state, err := pool.currentState()
+	if err != nil {
+		glog.Infof("Failed to get current state: %v", err)
+		return pending
+	}
```

Comment:
When can retrieving the current state fail? This type of silent error handling can really blow up. If there's a genuine error possibility, than that should be bubbled up through this method invocation and not (semi)silently logged and forgotten about.